File: src/logs.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

'''
    File name: logs.py
    Author: MathIsSimple
    Python Version: 3.7.0
    Disclaimer: I created this project to learn about custom encoding and python sockets,
                this projected isn't made to be used for maliscious intent. Do so at your own risk
'''

# Import needed core python modules

from sys import path

path.insert(0, './../lib/')

# Import needed custom python modules

from hasing      import decrypt
from fileManager import getFilesInDir
from fileManager import getFileExtension
from fileManager import fileExists

print("Open Logs")

log_dir = "./../logs/"

def start():
    logs = getFilesInDir(log_dir)

    for log in logs:
        print("Log File : " + log)

    log_to_open = input("File to Open : ")
    print("")
    if log_to_open == "END":
        exit(0)
    extension   = getFileExtension(log_to_open)

    if extension != ".txt":
        log_to_open = log_to_open + ".txt"

    log_to_open_location = log_dir+log_to_open

    if fileExists(log_to_open_location):
        f = open(log_to_open_location, "r")
        print("Opened log")

        while True:
            line = f.readline().replace("\n", "").replace("\r", "")
            if line == "": break
            print(decrypt(line))
        
        f.close()
    else:
        print("This log doesn't exist")
    print("")
    start()
start()

File: server/logs.js
/*

    File name: logs.js
    Author: MathIsSimple
    Using: Nodejs
    Type: Build
    Build Version: 0.7
    Disclaimer: I created this project to learn about custom encoding and python sockets,
                this projected isn't made to be used for maliscious intent. Do so at your own risk

*/

const fs       = require('fs');
const readline = require('readline');

const LOGS  = "./../logs/";

if (fs.existsSync(LOGS) == false) {
    fs.mkdirSync(LOGS);
}

fs.readdir(LOGS, (err, logs) => {
    if (err) {
        throw err;
    }
    console.log("");
    console.log("The Logs Are : ");
    console.log("");
    for (log of logs) {
        console.log("    "+log);
    }
    console.log("");
    ask("Wich Log Do You Want To Open : ", (log) => {
        fs.readFile(LOGS + log, (err, data) => {
            if (err) {
                throw err;
            }

            console.log("");

            data = JSON.parse(data);
            info = data.info;

            for (el of info) {
                console.log(el);
            }

            console.log("");

            for (let i = 0; i < data.commands.length; i ++) {
                console.log(data.commands[i]);
                console.log("");
                for (let j = 0; j < data.responses[i].length; j ++) {
                    console.log(data.responses[i][j]);
                }
                console.log("");
            }

        });
    });
});

function ask(question, callback) {
    let rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });
    rl.question(question, (command) => {
        rl.close();
        callback(command);
    });
}